### PR TITLE
Add stream related javadoc

### PR DESCRIPTION
**Pull request by npawar**
Adding some java doc to the interfaces
Moving SimpleConsumerFactory to KafkaConsumerFactory, and deprecating SimpleConsumerFactory

### CODE REVIEW

File: pinot-core/src/main/java/com/linkedin/pinot/core/realtime/stream/StreamLevelConsumer.java [hunk 1]
```diff
@@ -18,50 +18,57 @@
...
   /**
-   *
+   * Initialize and start the stream level consumer
+   * @throws Exception
    */
   void start() throws Exception;
 
   /**
-   *
+   * Set the offset to consume from
```
Comment (mcvsubbu) [inline pinot-core/src/main/java/com/linkedin/pinot/core/realtime/stream/StreamLevelConsumer.java]: Please remove this method from the interface

File: pinot-core/src/main/java/com/linkedin/pinot/core/realtime/stream/StreamLevelConsumer.java [hunk 2]
```diff
@@ -18,50 +18,57 @@
...
-   * return GenericRow
+   * Get next row from the stream and decode it into a generic row
+   * @param destination
+   * @return
    */
   GenericRow next(GenericRow destination);
 
   /**
-   *
+   * Get decoded row from the stream at a given offset
```
Comment (mcvsubbu) [inline pinot-core/src/main/java/com/linkedin/pinot/core/realtime/stream/StreamLevelConsumer.java]: Remove

File: pinot-core/src/main/java/com/linkedin/pinot/core/realtime/stream/StreamLevelConsumer.java [hunk 3]
```diff
@@ -18,50 +18,57 @@
...
    * @param offset
    * @return
    */
   GenericRow next(long offset);
 
   /**
-   *
+   * Get the current offset for the stream
```
Comment (mcvsubbu) [inline pinot-core/src/main/java/com/linkedin/pinot/core/realtime/stream/StreamLevelConsumer.java]: Remove

File: pinot-core/src/main/java/com/linkedin/pinot/core/realtime/stream/StreamLevelConsumer.java [hunk 4]
```diff
@@ -18,50 +18,57 @@
...
 
   /**
-   *
+   * Checkpoint the stream at the current offset
    */
   void commit();
 
   /**
-   *
+   * Checkpoint the stream at the given offset
```
Comment (mcvsubbu) [inline pinot-core/src/main/java/com/linkedin/pinot/core/realtime/stream/StreamLevelConsumer.java]: Remove

**Comment by npawar** [comment]
Removed all unused methods from StreamLevelConsumer

File: pinot-core/src/main/java/com/linkedin/pinot/core/realtime/stream/StreamLevelConsumer.java [hunk 5]
```diff
@@ -18,50 +18,32 @@
...
-  /**
-   *
-   * @return
-   */
-  long currentOffset();
+  GenericRow next(GenericRow destination);
 
   /**
-   *
+   * Commit the offsets
```
Comment (mcvsubbu) [inline pinot-core/src/main/java/com/linkedin/pinot/core/realtime/stream/StreamLevelConsumer.java]: Please add a comment here that the next call to consume should exclude all events consumed until the commit() method is called, and start from newer events not yet consumed

**Comment by npawar** [comment]
Addressed review comments @mcvsubbu

**Pull request merged**